components/camel-tahu/src/test/java/org/apache/camel/component/tahu/SparkplugTCKService.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.tahu;
+
+import java.nio.charset.StandardCharsets;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.NotifyBuilder;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.component.tahu.services.HiveMQService;
+import org.apache.camel.component.tahu.services.HiveMQServiceFactory;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.support.builder.PredicateBuilder;
+import org.apache.camel.test.infra.common.services.TestService;
+import org.apache.camel.test.infra.core.MockUtils;
+import org.eclipse.paho.client.mqttv3.IMqttMessageListener;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import static org.apache.camel.test.junit5.TestSupport.bodyAs;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@Testcontainers
```

Comment:
A few notes:

1. We don't use those annotations.
2. Container-based test infrastructure should be in the [test-infra modules](https://camel.apache.org/manual/test-infra.html).
3. The base service should be an interface and the concrete test service an implementation of that interface.
4. Ideally, it should be a singleton service, so it doesn't overload ASF CI.